tt_metal/hw/ckernels/grayskull/metal/common/metal_ckernel_globals.h

```diff
@@ -0,0 +1,63 @@
+// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+//TODO: This file should be deleted after fixing redefinition errors,
+// functions should be moved to ckernel_globals.h
+#pragma once
+
+#include <cstdint>
+#include "ckernel_structs.h"
+#include "risc_attribs.h"
+#include "tensix_functions.h"
+#include "hostdevcommon/common_runtime_address_map.h"
+
+extern uint32_t __ldm_bss_start[];
+extern uint32_t __ldm_bss_end[];
+extern uint32_t __ldm_data_start[];
+extern uint32_t __ldm_data_end[];
+extern void (* __init_array_start[])();
+extern void (* __init_array_end[])();
+extern uint32_t __firmware_start[];
+
+extern void kernel_init();
+extern void kernel_launch();
+
+inline void l1_to_local_mem_copy(uint32_t *local_mem_addr, uint32_t tt_l1_ptr *l1_addr, int32_t len) {
+    // Cover L1 load latency of 6 cycles for the bulk of the copy
+    int32_t n = 0;
+    while (n < len - 5) {
+        uint32_t v0 = l1_addr[n + 0];
+        uint32_t v1 = l1_addr[n + 1];
+        uint32_t v2 = l1_addr[n + 2];
+        uint32_t v3 = l1_addr[n + 3];
+        uint32_t v4 = l1_addr[n + 4];
+        uint32_t v5 = l1_addr[n + 5];
+        local_mem_addr[n + 0] = v0;
+        local_mem_addr[n + 1] = v1;
+        local_mem_addr[n + 2] = v2;
+        local_mem_addr[n + 3] = v3;
+        local_mem_addr[n + 4] = v4;
+        local_mem_addr[n + 5] = v5;
+        n += 6;
+    }
```

Comment:
#pragma unroll(5)
for (int32_t n = 0; n < len - 5; n++) {
 ...
}

for (; n < len; n++) {
 ...
}